contracts/dao/neutron-flashloans/src/contract.rs

```diff
@@ -0,0 +1,403 @@
+) -> Result<Vec<Coin>, ContractError> {
+    let mut fee: Vec<Coin> = Vec::with_capacity(requested_amount.len());
+    for coin in requested_amount {
+        let coin_fee = Coin::new((fee_rate * coin.amount).u128(), coin.denom);
+        fee.push(coin_fee)
+    }
+
+    Ok(fee)
+}
+
+// Sums the pre_loan_balances with the fee.
+// WARNING: this function assumes that the input vectors are of the same length,
+// and that the order of the denoms is the same.
+fn calculate_expected_balances(
+    pre_loan_balances: Vec<Coin>,
+    fee: Vec<Coin>,
+) -> Result<Vec<Coin>, ContractError> {
+    let mut expected_balances: Vec<Coin> = Vec::with_capacity(pre_loan_balances.len());
+    for (index, coin) in pre_loan_balances.iter().enumerate() {
+        expected_balances.push(Coin::new(
+            coin.amount.checked_add(fee[index].amount)?.u128(),
+            coin.denom.clone(),
+        ))
+    }
+
+    Ok(expected_balances)
+}
+
+/// A simple function to build the (stargate -> authz -> bank) /cosmos.bank.v1beta1.MsgSend message.
+fn get_stargate_authz_bank_send_msg(
+    env: Env,
+    config: Config,
+    borrower: Addr,
+    amount: Vec<Coin>,
+) -> CosmosMsg<NeutronMsg> {
+    // First we create the bank MsgSend
+    let bank_send_msg = MsgSend {
+        from_address: config.source.to_string(),
+        to_address: borrower.to_string(),
+        amount: amount
+            .iter()
+            .map(|x| ProtoCoin {
+                denom: x.clone().denom,
+                amount: x.amount.to_string(),
+            })
+            .collect(),
+    };
+
+    // Then we wrap it in an authz message
+    let authz_msg_exec = MsgExec {
+        grantee: env.contract.address.to_string(),
+        msgs: vec![Any {
+            type_url: BANK_MSG_SEND_TYPE_URL.to_string(),
+            value: bank_send_msg.encode_to_vec(),
+        }],
+    };
+
+    // Then we wrap the authz message in a stargate message, because there is
+    // no custom support for authz messages in CosmWasm.
+    let stargate_authz_msg_exec: CosmosMsg<NeutronMsg> = CosmosMsg::Stargate {
+        type_url: AUTHZ_MSG_EXEC_TYPE_URL.to_string(),
+        value: Binary(authz_msg_exec.encode_to_vec()),
+    };
+
+    stargate_authz_msg_exec
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
+    match msg.id {
+        AUTHZ_BANK_SEND_REPLY_ID => {
+            // If we are here, the money is already on the borrower's account. This means that
+            // we can proceed to call the borrower's handler.
+
+            let config: Config = CONFIG.load(deps.storage)?;
+            let active_loan = must_get_active_loan(&deps)?;
+
+            let msg = CosmosMsg::Wasm(WasmMsg::Execute {
+                contract_addr: active_loan.borrower.to_string(),
+                msg: to_json_binary(&BorrowerInterface::ProcessLoan {
+                    // We sent the return address to the source address
```

Comment:
```suggestion
                    // We set the return address to the source address
```
btw is this comment even necessary? it just literally explains the field assignment operation saying nothing about the motivation (which must be obvious)